src/config.rs

```diff
@@ -751,6 +754,14 @@ fn validate_config(cfg: &ConfigItems) -> Result<(), Error> {
         )
     }
 
+    const KNOWN_FLAGS: &[&str] =
+        &["autofill-overlay", "autofill-v2", "browser-fileless-import", "fido2-vault-credentials"];
+    for flag in parse_experimental_client_feature_flags(&cfg.experimental_client_feature_flags).keys() {
+        if !KNOWN_FLAGS.contains(&flag.as_str()) {
+            err!(format!("The experimental client feature flag {flag:?} is unrecognized. Please ensure the feature flag is spelled correctly and that it is supported in this version."));
```

Comment:
```suggestion
            warn!("The experimental client feature flag {flag:?} is unrecognized. Please ensure the feature flag is spelled correctly and that it is supported in this version.");
```

I don't think the validation should return an error, if we ever update the allowed flags and remove some, we don't want that to break all the users that have enabled it before.

This could potentially happen if we find a big error in the implementation of one, or also when the feature flag code is stabilized, the flag might be deleted once all the clients support it.

Reply:
@dani-garcia I'm not sure I agree. I think it's better to return an error so that users can't enable unsupported flags (or at least immediately notice that they have misspelled a feature flag when changing the flag via the admin panel - which I'm not sure would happen if this were a mere warning) and if we remove a feature flag from `KNOWN_FLAGS` (because it's not supported anymore or was added prematurely) I'd probably want this validation to fail. Furthermore I don't think that it's an issue enabling feature flags that are not in use anymore (like `display-kdf-iteration-warning`), so we won't have to remove them as soon as possible. (Users might also want to clean up removed feature flags that they have enabled to reduce the size of the returned `/api/config` - without immediate feedback I don't think this will happen.)

Reply:
In order not to break too many installations we probably should not remove features from `KNOWN_FLAGS` that we have enabled by default (like `fido2-vault-credentials`) or only if there's an important reason to do so or we really want to remove it. (I think the other flags which we don't enable by default - would be fair game as users would be responsible if they change the defaults).